runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -392,11 +409,28 @@ function M.enable(enable, filter)
   end
 
   vim.validate({ enable = { enable, 'boolean', true }, filter = { filter, 'table', true } })
+  enable = enable == nil or enable
   filter = filter or {}
-  if enable == false then
-    _disable(filter.bufnr)
+
+  if filter.bufnr == nil then
+    globalstate.enabled = enable
+    for bufnr, _ in pairs(bufstates) do
+      if api.nvim_buf_is_loaded(bufnr) then
+        if enable == false then
+          _disable(bufnr)
+        else
+          _enable(bufnr)
+        end
```

Comment:
Looks like this erases all the buffer-local states. That seems to match `vim.diagnostic.enable()`: https://github.com/neovim/neovim/blob/e778e0116198470ba037b9426f4ff7fa5cb7f880/runtime/lua/vim/diagnostic.lua#L2046-L2053

That seems a bit "destructive". I tend to think that global enable() should act like a gate: 

- if globally enabled, then defer to the buffer-local flag
- if globally disabled, don't check the buffer-local flag

But since this matches `vim.diagnostic.enable()` I think we can go with this for now.